### PR TITLE
STCLI-190 Update dependencies (CVE-2021-3807)

### PR DESCRIPTION
**Pull request by zburke**
Some out of date dependencies contain security vulnerabilities:

* ansi-regex - Regular Expression Denial of Service (ReDoS) - https://nvd.nist.gov/vuln/detail/CVE-2021-3807. Vulnerable versions include 3.0.0 and 4.1.0; versions included after this patch include 5.0.1, 3.0.1, and 2.1.1.
* unset-value - Prototype Pollution - https://app.snyk.io/vuln/SNYK-JS-UNSETVALUE-2400660. Still present as a transitive dep via kopy and @folio/stripes-webpack. @julianladisch, do you recall if your original PR dealt with this somehow, or were you just mentioning it for completeness?
* istanbul-reports - Reverse Tabnabbing - https://app.snyk.io/vuln/SNYK-JS-ISTANBULREPORTS-2328088. Vulnerable versions are < 3.1.3; this patch bumps to 3.1.4.
* ejs - Arbitrary Code Injection - https://app.snyk.io/vuln/SNYK-JS-EJS-1049328. Vulnerable versions are < 3.1.6; this patch removes this transitive dependency of kopy v8 bumping to v9.

This PR replaces #280, which was reverted in #287 because we missed changes to the `kopy` API between v8 and v9. 

Refs [STCLI-190](https://issues.folio.org/browse/STCLI-190)

### CODE REVIEW

File: lib/stripes-cli.js [hunk 1]
```diff
@@ -21,19 +22,9 @@ updateNotifier({
   message: 'Update available - Refer to README.md:\nhttps://github.com/folio-org/stripes-cli',
 });
 
-// Monkey-patch Yargs to inform user of an unrecognized command when no suggestions could be provided
```
Comment (ryandberger) [inline lib/stripes-cli.js:24]: I know this is incompatible with new versions of `yargs` but the downside to removing this is that if you type a command that `yargs` can't make a "did you mean?" suggestion, the CLI will just quietly exit and do nothing. My vote is that it's worth dropping this functionality to move forward (and perhaps file an enhancement request to `yargs`).

File: package.json [hunk 2]
```diff
@@ -35,38 +35,38 @@
...
     "rimraf": "^2.6.2",
     "semver": "^5.6.0",
     "simple-git": "^3.5.0",
     "supports-color": "^4.5.0",
-    "update-notifier": "^2.3.0",
+    "update-notifier": "^5.1.0",
     "webpack": "~5.68.0",
     "webpack-bundle-analyzer": "^4.4.2",
-    "yargs": "^13.1.0"
+    "yargs": "^17.3.1"
```
Comment (ryandberger) [inline package.json:69]: Loving these much needed updates 😄

**Review by ryandberger — APPROVED** [review]

**Review by JohnC-80 — APPROVED** [review]

**Comment by julianladisch** [comment]
> unset-value - Prototype Pollution - https://app.snyk.io/vuln/SNYK-JS-UNSETVALUE-2400660. Still present as a transitive dep via kopy and @folio/stripes-webpack.

`yarn why unset-value` doesn't show any transitive dep via kopy for this pull request. Upgrading kopy from 8.2.3 to 9.4.2 removed this transitive dep. kopy 9.4.2 is advertised as "Zero-dependency": https://www.npmjs.com/package/kopy

`@folio/stripes-webpack` v3.0.3 uses unset-value, however, on `@folio/stripes-webpack` master it has been removed. Therefore upgrading stripes-webpack is blocked and excluded from this pull request.

**Review by julianladisch — APPROVED** [review]
Looks good, thanks!

**Comment by zburke** [comment]
You're right, @julianladisch. `unset-value` came in as a transitive dep in several places, all of which have been updated; it is no longer present. I must have been working with an outdated `yarn.lock` file or `node_modules` directory with cruft leftover from a previous build.

**Pull request merged**